arkfilepicker/src/main/java/space/taran/arkfilepicker/presentation/folderstree/FoldersTreeView.kt

```diff
@@ -0,0 +1,170 @@
+package space.taran.arkfilepicker.presentation.folderstree
+
+import android.util.Log
+import androidx.recyclerview.widget.RecyclerView
+import com.mikepenz.fastadapter.FastAdapter
+import com.mikepenz.fastadapter.GenericItem
+import com.mikepenz.fastadapter.adapters.ItemAdapter
+import com.mikepenz.fastadapter.diff.FastAdapterDiffUtil
+import java.nio.file.Path
+
+class FolderTreeView(
+    private val rv: RecyclerView,
+    private val onNavigateClick: (FolderNode) -> Unit,
+    private val onAddClick: (FolderNode) -> Unit,
```

Comment:
Callbacks `onNavigate` and `onAdd` look like Navigator-specific logic.

Reply:
Yes, I want to reuse this component

Reply:
Maybe it's an interesting idea.. I just imagine that this filepicker will be in several apps, e.g. a user invokes it from Retouch for saving a picture. It might be useful to go inside a folder with Navigator before selecting the folder for saving a picture (just to make sure it's a correct folder). Not sure about creating new favorite folder for saving the picture there.

But when we'll be solving [this issue](https://github.com/ARK-Builders/ark-filepicker/issues/7), we'll need to separate navigator logic better. Maybe with some configuration of callbacks.